### PR TITLE
Refactor GitHub Action for Emulator Runner Job Redundancy

### LINKED ISSUE #3615: Refactor GitHub Action for Emulator Runner Job Redundancy
**Describe the bug**
The current GitHub Actions workflow contains duplicated code for creating instrumentation coverage with the Android Emulator Runner. The only difference is in the configuration for api-level 33, where the target, ram-size, sdcard-path-or-size, and heap-size parameters need adjustments.

**Proposed Solution**
Refactor the code to eliminate redundancy and simplify the workflow by using a single job definition with conditional parameter assignments based on the value of `matrix.api-level`.

### PR DESCRIPTION
**Pull request by gouri-panda**
Fixes #3615 

**Description**
This pull request addresses the redundancy in the GitHub Actions workflow for creating instrumentation coverage with the Android Emulator Runner. The current workflow contains duplicated code, and this PR refactors it into a single job definition with conditional parameter assignments based on the value of matrix.api-level.

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (MohitMaliFtechiz) [inline .github/workflows/ci.yml:50]: @gouri-panda Since `ram-size` is the same for both API levels, we should get rid of this condition here.

**Review by MohitMaliFtechiz — CHANGES_REQUESTED** [review]
@gouri-panda Thanks for the PR, just need a small change, and then we are good to merge.

**Review by MohitMaliFtechiz — APPROVED** [review]
@gouri-panda Thanks LGTM, once CI paas we can merge.

**Review by kelson42 — CHANGES_REQUESTED** [review]
Indentation is incorrect

**Comment by gouri-panda** [comment]
@kelson42 Thanks!

**Review by MohitMaliFtechiz — APPROVED** [review]
@gouri-panda Thanks.

**Pull request merged**
